src/wallet/wallet_rpc_server.cpp

```diff
@@ -2431,6 +2431,97 @@ bool wallet_rpc_server::on_create_wallet(const wallet_rpc::COMMAND_RPC_CREATE_WA
 	return true;
 }
 //------------------------------------------------------------------------------------------------------------------------------
+bool wallet_rpc_server::on_restore_wallet(const wallet_rpc::COMMAND_RPC_RESTORE_WALLET::request &req, wallet_rpc::COMMAND_RPC_RESTORE_WALLET::response &res, epee::json_rpc::error &er)
+{
+	if(m_wallet_dir.empty())
+	{
+		er.code = WALLET_RPC_ERROR_CODE_NO_WALLET_DIR;
+		er.message = "No wallet dir configured";
+		return false;
+	}
+
+	std::string wallet_file = m_wallet_dir + "/" + req.filename;
+
+	std::string seed = req.seed;
+
+	std::vector<std::string> wseed;
+	boost::algorithm::trim(seed);
+	boost::split(wseed, req.seed, boost::is_any_of(" "), boost::token_compress_on);
+
+	std::string language;
+	crypto::secret_key_16 seed_14;
+	uint8_t seed_extra;
+	crypto::secret_key seed_25;
+	bool decode_14 = false, decode_25 = false;
+	if(wseed.size() == 12 || wseed.size() == 14)
+	{
+		decode_14 = crypto::Electrum14Words::words_to_bytes(req.seed, seed_14, seed_extra, language);
+	}
+	else if(wseed.size() >= 24 && wseed.size() <= 26)
+	{
+		decode_25 = crypto::Electrum25Words::words_to_bytes(req.seed, seed_25, language);
+	}
+	else
+	{
+		er.code = WALLET_RPC_ERROR_CODE_UNKNOWN_ERROR;
+		er.message = "Unkown seed size " + std::to_string(wseed.size()) + " please enter 12, 14, 24, 25 or 26 words";
+		return false;
+	}
+
+	if(!decode_14 && !decode_25)
+	{
+		er.code = WALLET_RPC_ERROR_CODE_UNKNOWN_ERROR;
+		er.message = "Electrum-style word list failed verification";
+		return false;
+	}
+
+	namespace po = boost::program_options;
+	po::variables_map vm2;
+
+	{
+		po::options_description desc("dummy");
+		const command_line::arg_descriptor<std::string, true> arg_password = {"password", "password"};
+		const char *argv[4];
+		int argc = 3;
+		argv[0] = "wallet-rpc";
+		argv[1] = "--password";
+		argv[2] = req.password.c_str();
```

Comment:
Only a note: This construct is risky if `req` is destroyed and `argv` is used after that point the pointer in `argv[2]` would be invalid. This is not the case in this code snipped but I would point to this risky snipped.